packages/epo-react-lib/src/layout/SlideoutMenu/MenuItem/MenuItem.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+  FunctionComponent,
+  PropsWithChildren,
+  useContext,
+  useRef,
+  useEffect,
+} from "react";
+import MenuContext from "@/contexts/Menu";
+import { StyledMenuItem, StyledMenuItemWrapper } from "./styles";
+import { ButtonProps } from "@/atomic/Button";
+
+interface MenuItemProps extends ButtonProps {
+  text: string;
+}
+
+const MenuItem: FunctionComponent<PropsWithChildren<MenuItemProps>> = ({
+  text,
+  children,
+  ...buttonProps
+}) => {
+  const menuContext = useContext(MenuContext);
+  const menuItemRef = useRef<HTMLButtonElement>();
+
+  if (!menuContext) {
+    throw new Error("Menu item must be used within a Menu Context");
+  }
+
+  const { menuItems, currentIndex } = menuContext;
+
+  useEffect(() => {
```

Comment:
`menuItem` adds itself to the menuItem context and figures out if it is currently active or not